opentelemetry-sdk/src/trace/provider.rs

```diff
@@ -36,36 +90,60 @@ static NOOP_TRACER_PROVIDER: Lazy<TracerProvider> = Lazy::new(|| TracerProvider
             span_limits: SpanLimits::default(),
             resource: Cow::Owned(Resource::empty()),
         },
+        is_shutdown: AtomicBool::new(true),
```

Comment:
Why have we initialized the no-op tracer provider as an already shutdown provider?

I know this wouldn't make much difference in functionality but semantically it would be weird if I call shutdown on the global provider and get an error saying it has already been shut down.

Reply:
global [shutdown_tracer_provider ](https://github.com/lalitb/opentelemetry-rust/blob/ea4b5e4d992610bb23a059f05253d0c47e262015/opentelemetry/src/global/trace.rs#L421) naming is bit confusing. It doesn't invoke shutdown, only decrements the TracerProviderInner reference. Once all the tracers are dropped, the shutdown will get invoked through `Drop`.

And the user will never get hold of NOOP_TRACER_PROVIDER instance. It's just internally used to create the tracer instance after shutdown.

Reply:
Okay so then we could initialize no-op tracer provider with `is_shutdown` as `false`?

Reply:
> Okay so then we could initialize no-op tracer provider with is_shutdown as false?

Making it `false` fails this test - https://github.com/open-telemetry/opentelemetry-rust/blob/ea4b5e4d992610bb23a059f05253d0c47e262015/opentelemetry-sdk/src/trace/provider.rs#L531
And this test make me realise user can access it through tracer.provider(), and can try to invoke shutdown on it. But feel free to raise an issue if there are any improvements here.